pkg/lib/scoped/syncer.go

```diff
@@ -74,6 +78,19 @@ func (s *UserDefinedServiceAccountSyncer) SyncOperatorGroup(in *v1.OperatorGroup
 	// A service account has been specified, we need to update the status.
 	sa, err := s.client.KubernetesInterface().CoreV1().ServiceAccounts(namespace).Get(context.TODO(), serviceAccountName, metav1.GetOptions{})
 	if err != nil {
+		// Set OG's status condition to indicate SA is not found
+		cond := metav1.Condition{
+			Type:    v1.OperatorGroupServiceAccountCondition,
+			Status:  metav1.ConditionTrue,
+			Reason:  v1.OperatorGroupServiceAccountReason,
+			Message: fmt.Sprintf("ServiceAccount %s not found", serviceAccountName),
+		}
+
+		meta.SetStatusCondition(&in.Status.Conditions, cond)
+		_, err = s.update(in, nil)
```

Comment:
This actually overrides the Get call error, which is then used on line 94, so the error returned by the function may be nil or refer to an issue setting the Condition.

Reply:
I already corrected this. The condition is added when SA is not found (according to the error). When there is a different type of error, then log and return that error.